src/librustc_driver/driver.rs

```diff
@@ -1650,17 +1649,14 @@ pub fn build_output_filenames(
             // "-" as input file will cause the parser to read from stdin so we
             // have to make up a name
             // We want to toss everything after the final '.'
-            let dirpath = match *odir {
-                Some(ref d) => d.clone(),
-                None => PathBuf::new(),
-            };
+            let dirpath = (*odir).as_ref().cloned().unwrap_or_else(|| PathBuf::new());
```

Comment:
`unwrap_or_else(PathBuf::new);`

Reply:
Or `unwrap_or_default` that I forgot existed ^^.